tests/cases/employee_health_portal/case_employee_health_portal_request_appointment.yaml

```diff
@@ -75,50 +67,170 @@ TestEmployeeHealthPortalRequestAnAppointmentForUser:
+      Id: $PAGE.employee_health_home_page.calendar_day_button$
+    - Type: sleep
+      Time: 1 
+    # Click on the View upcoming appointments
     - Type: wait_for
       Strategy: xpath
-      Id: $PAGE.employee_health_telehealth_request_form.calendar_month_view$
+      Id: $PAGE.employee_health_home_page.view_upcoming_appointment_button$
     - Type: click
       Strategy: xpath
-      Id: $PAGE.employee_health_telehealth_request_form.calendar_day_button$
+      Id: $PAGE.employee_health_home_page.view_upcoming_appointment_button$
+    # getting second day
+    - Type: get_attribute
+      Strategy: xpath
+      Id: $PAGE.employee_health_home_page.second_day$
+      Greps:
+        - var: SECOND_SCHEDULED_DAY
+          attr: datetime
+          condition: nempty
+          match: "(.*)"
+    - Type: get_day
+      Timestamp: $AND_CLI_SECOND_SCHEDULED_DAY$
+      ResultVar: DAY
+    # checking second day
     - Type: click
       Strategy: xpath
-      Id: $PAGE.employee_health_telehealth_request_form.calendar_hour_button$
+      Id: $PAGE.employee_health_home_page.calendar_day_button$
+    # look for another day without even scheduled a date.
+    - Type: generate_random_day
+      ResultVar: Day
+    # checking a RandomDay
     - Type: click
       Strategy: xpath
-      Id: $PAGE.employee_health_telehealth_request_form.create_event_button$
-  # Verify if the appointment is showing on the list
+      Id: $PAGE.employee_health_home_page.calendar_day_button$
+    - Type: sleep
+      Time: 1
     - Type: wait_for
       Strategy: xpath
-      Id: $PAGE.employee_health_telehealth_request_form.appointment_list$
+      Id: $PAGE.employee_health_home_page.view_upcoming_appointment_button$
+    - Type: click
+      Strategy: xpath
+      Id: $PAGE.employee_health_home_page.view_upcoming_appointment_button$
+    # getting third day scheduled
     - Type: get_attribute
       Strategy: xpath
-      Id: $PAGE.employee_health_telehealth_request_form.appointment_list_description$
+      Id: $PAGE.employee_health_home_page.third_day$
       Greps:
-        - var: SHOWED_DESCRIPTION
-          attr: textContent
+        - var: THIRD_SCHEDULED_DAY
+          attr: datetime
           condition: nempty
           match: "(.*)"
-    - Type: assert
-      Asserts:
-        - Type: eq
-          Var: SHOWED_DESCRIPTION
-          Value: $AND_CLI_PAIN_DESCRIPTION$
\ No newline at end of file
+    - Type: get_day
+      Timestamp: $AND_CLI_THIRD_SCHEDULED_DAY$
+      ResultVar: DAY
+    # checking third day
+    - Type: click
+      Strategy: xpath
+      Id: $PAGE.employee_health_home_page.calendar_day_button$
+    # look for another day without even scheduled a date.
+    - Type: generate_random_day
+      ResultVar: Day
+    # checking a RandomDay
+    - Type: click
+      Strategy: xpath
+      Id: $PAGE.employee_health_home_page.calendar_day_button$
+    - Type: sleep
+      Time: 1
```

Comment:
is this sleep needed

Reply:
yes, when I use wait_for  the test failure trying to click on View upcoming appointment button.